### PR TITLE
Version 2 of the r image

### LINKED ISSUE #51: pandoc upgrade
I'm using [Rmarkdown to create html reports](https://github.com/opensafely/comparative-ve-research/blob/748cb52e55cd7347ae6f90ee16c9cd4403e178a7/analysis/report/draft-manuscript.Rmd). These include citations to journal articles, using [a bibliography](https://github.com/opensafely/comparative-ve-research/blob/748cb52e55cd7347ae6f90ee16c9cd4403e178a7/analysis/report/references.bib). When I render the html locally using my own R installation (with pandoc version 2.11.4), everything works. When I try to render through the opensafely CLI (with pandoc version 1.19.2.4), it fails, with the following error message:

```bash
/usr/bin/pandoc +RTS -K512m -RTS draft-manuscript.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output /workspace/output/report/draft-manuscript.html --email-obfuscation none --self-contained --standalone --section-divs --template /usr/local/lib/R/site-library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --variable 'theme:bootstrap' --include-in-header /tmp/RtmpSd0tDk/rmarkdown-str852cf4586.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --filter pandoc-citeproc
     pandoc: Error running filter pandoc-citeproc:
     Could not find executable 'pandoc-citeproc'.
     Error: pandoc document conversion failed with error 83
     Backtrace:
     1: stop("pandoc document conversion failed with error ", result,
     2: pandoc_convert(input_files, pandoc_to, output_format$pandoc$from,
     3: convert(output_file, run_citeproc)
     4: rmarkdown::render("analysis/report/draft-manuscript.Rmd", knit_root_dir = "/workspace",
```

When I [remove the citations](https://github.com/opensafely/comparative-ve-research/commit/748cb52e55cd7347ae6f90ee16c9cd4403e178a7), it works again. 

Here's a similar issue https://github.com/crsh/papaja/issues/427#issuecomment-728805035, in particular: 

> I'm pretty sure this is a problem caused by the fact that the new pandoc version replaces `pandoc-citeproc` with `citeproc`

I'm hoping that an update of pandoc will therefore fix the problem, and not break other things 🤞🏻

### LINKED ISSUE #74: Add kableExtra package, please
I want to produce a publicaiton-ready table using the [kableExtra](https://www.rdocumentation.org/packages/kableExtra/versions/1.3.4) package in R. I previously requested installation of the [magick](https://github.com/opensafely-core/r-docker/issues/67) package because the developers of kable libraries [mentioned it as a dependency](https://haozhu233.github.io/kableExtra/save_kable_and_as_image.html)...but of course I never requested installation of kableExtra.

I haven't needed the other dependency ([webshot](https://www.rdocumentation.org/packages/webshot/versions/0.5.2/) package) to produce the publication-ready table when running a local instance of R (not in the OpenSAFELY docker). Would it be worth installing that too, or is its webpage functionality a No-No for containers?

### LINKED ISSUE #75: Suggestion: Install binary R packages from the public RSPM repo
In case helpful - apologies if you've already thought about or are already doing this.

The following is what alot of the GitHub Actions for R packages are using to install binary (instead of source) packages on Linux. You can see what I describe below being set in **r-lib/actions** [here](https://github.com/r-lib/actions/blob/f38217db7df21f0dc51e542644737645fd0b2d47/setup-r/src/installer.ts#L495).

Binary versions of R packages for Linux distros are available using the public RStudio Package Manager (RSPM) repo `https://packagemanager.rstudio.com/all/__linux__/bionic/latest` URL (or the URL with the Ubuntu Codename and/or date you prefer) as your CRAN repo URL. 

The general details are [here](https://packagemanager.rstudio.com/client/#/repos/1/overview). 

The trick **r-lib/actions** uses, and that anyone can use, is that you don't have to use the RStudio Package Manager to use this public repo. So the easiest way to use this is to call `install.packages()`/`renv::install()` with the `repos` argument, e.g.

```r
install.packages("ggplot2", repos = "https://packagemanager.rstudio.com/all/__linux__/bionic/latest")
# or: renv::install() with same arguments
```

(There are alternative ways to change `options()$repos["CRAN"]`, e.g. using an `Rprofile.site` file.)

## System prerequisites

With RSPM binary packages there are occasionally some additional Linux system requirements required (I guess for when the package actually runs and perhaps for when they load). RStudio provide a shell script for each package on their webpage for each package, e.g. for **magick** 

2022-03-14 15_42_00-RStudio Package Manager

## Result

Then the Dockerfile could read in the list of packages in `packages.txt` and would run substantially faster. It should be practical to build the Dockerfile for every new package added. 

And this should make it easier to update the version of R and the versions of the packages included because the packages would update everytime the container was built unless you use a URL with a fixed date.

### LINKED ISSUE #88: survcomp package
Could you please install survcomp package? http://www.bioconductor.org/packages/release/bioc/html/survcomp.html

Thanks!

### LINKED ISSUE #101: Possible next steps for r-docker
Here are some possible next steps for r-docker, following a meeting with @bloodearnest, @milanwiedemann, and @rebkwok. They're arranged from least to most effort, over the short-to-medium term.

**Continue to update the existing image.** Address:

- #99
- #88
- #74 
- #51 
- #36 

**Fix the image.**

- https://github.com/opensafely/reverse-engineer-r-docker

**Create a new - and reversion the existing - image.**

- The existing image could become `r:legacy`; the new image could become `r:latest`
- Automatically open PRs for all studies that use the image, `s/r:latest/r:legacy/`

**Handle dependencies better.**

- We could create a volume for dependencies
- We could create a private, limited archive network
- We could create tooling to help users bundle dependencies with their studies (e.g. `opensafely add-dependency ...`)

### LINKED ISSUE #160: sjPlot package
https://cran.r-project.org/web/packages/sjPlot/index.html

Please could sjPlot be added to support analysis to explore the impact of demographic characteristics on end of life care? We are attempting to use the MAIHDA approach - Intersectional Multilevel Analysis of Individual Heterogeneity and Discriminatory Accuracy (MAIHDA). Thank you.

### LINKED ISSUE #170: update dtplyr version
The `dtplyr` package lets R users do super-fast `data.table` operations within the `dplyr` API. Since `dplyr` is currently much more widely used in OpenSAFELY than `data.table`, getting `dplyr` users to switch to `dtplyr` for intensive operations could save a lot of time and memory. 

The `dtplyr` R package is currently installed at version `1.0.1`. The latest version is `1.3.1` which includes support for a much broader set of data wrangling functions, including pivoting (eg, wide format to long format) which is expensive. An update would be great!

package website: https://dtplyr.tidyverse.org/
CRAN: https://cran.r-project.org/package=dtplyr

### PR DESCRIPTION
**Pull request by remlapmot**
Closes #51 
Because pandoc is updated in v2 image.

Closes #74
The kableExtra package is included but we have chosen not to include phantomjs, which we think has gone unmaintained for a while.

Closes #75 
As the v2 image installs binary R packages for Linux from the Posit Public Package Manager (formerly known as RStudio Package Manager).

Closes #88
Because survcomp is a Bioconductor package and we have currently chosen not to include Bioconductor packages.

Closes #101 
As this is the next step for r-docker.

Closes #160 
Adds sjPlot to v2 image.

Closes #170 
The dtplyr package is included at its current version.

### CODE REVIEW

**Review by bloodearnest — APPROVED** [review]
Let's gooooooo!
